modules/Microsoft.ContainerInstance/containerGroups/deploy.bicep

```diff
@@ -49,6 +102,46 @@ var identity = identityType != 'None' ? {
   userAssignedIdentities: !empty(userAssignedIdentities) ? userAssignedIdentities : null
 } : null
 
+var dnsConfig = !empty(dnsNameServers) ? {
+  nameServers: dnsNameServers
+  searchDomains: dnsSearchDomains
+} : null
+
+var encryptionProperties = !empty(encryptionVaultBaseUrl) ? {
+  vaultBaseUrl: encryptionVaultBaseUrl
+  keyName: encrytionKeyName
+  keyVersion: encryptionKeyVersion
+} : null
+
+var subnetIds = !empty(subnetId) ? [
+  {
+    id: subnetId
+  }
+] : null
+
+var generatedDomainNameLabelScope = !empty(dnsNameServers) ? autoGeneratedDomainNameLabelScope : null
+
+var basicContainerProperties = {
+  containers: containers
+  dnsConfig: dnsConfig
+  encryptionProperties: encryptionProperties
+  imageRegistryCredentials: imageRegistryCredentials
+  initContainers: initContainers
+  restartPolicy: restartPolicy
+  osType: osType
+  ipAddress: {
+    type: ipAddressType
+    autoGeneratedDomainNameLabelScope: generatedDomainNameLabelScope
+    dnsNameLabel: dnsNameLabel
+    ports: ipAddressPorts
+  }
+  sku: sku
+  subnetIds: subnetIds
+  volumes: volumes
+}
+
+var containerProperties = !empty(dnsNameServers) ? union(basicContainerProperties, { dnsConfig: dnsConfig }) : basicContainerProperties
```

Comment:
Why not do this directly int the `basicContainerProperties` var?

Also, I'm personally usually not a big fan of moving the entire `properties` away from the object they belong to, as it can become difficult to track down what actually is happening.  But that's more my personal taste and debatable.

Reply:
It is ugly, agree but `dnsConfig` as a null object is blocked bei the azure API.

Reply:
done

Reply:
@AlexanderSehr in this case, we have to many parameters where `null` is not allowed. The result is a Union of 10 objects and this is not nice to read.

so lets go with the variable option.